sdk/src/main/scala/io/horizen/history/AbstractHistory.scala

```diff
@@ -167,7 +167,7 @@ abstract class AbstractHistory[
     if(newChainSuffix.isEmpty && currentChainSuffix.isEmpty)
       throw new IllegalArgumentException("Cannot retrieve fork changes. Fork length is more than params.maxHistoryRewritingLength")
 
-    val newChainSuffixValidity: Boolean = !newChainSuffix.tail.map(isSemanticallyValid)
+    val newChainSuffixValidity: Boolean = params.resetModifiersStatus || !newChainSuffix.tail.map(isSemanticallyValid)
```

Comment:
Maybe it is a good idea to put a comment here, just as a reminder of why there is this flag check here